go/vt/sqlparser/ast.go

```diff
@@ -2710,6 +2710,11 @@ type (
 		YCordinate Expr
 	}
 
+	//LineString represents LineString(POINT(x,y), POINT(x,y), ..) expression
+	LineStringExpr struct {
+		PointParams []*PointExpr
```

Comment:
A `LineString()` function only takes point values as arguments, but these could come from various sources. You could use column values, user defined variables, or other functions that return `point` values. I don't think we can hard code these arguments to `PointExpr`

```sql
mysql> create table geom (pt1 point, pt2 point);
Query OK, 0 rows affected (0.01 sec)

mysql> select linestring(pt1, pt2) from geom;
```

☝️ this is a small example showing what I mean.

My suggestion is to use `Expr` instead of `*PointExpr`

Reply:
That makes a lot of sense. I have updated yacc grammar to accept ``expression_list`` instead of ``point_expression_list``, adding a test case to show that the example that u gave gets parsed now.